c10/core/Device.cpp

```diff
@@ -47,9 +47,17 @@ DeviceType parse_type(const std::string& device_string) {
   if (device != types.end()) {
     return device->second;
   }
+  std::vector<const char*> device_names;
```

Comment:
Can `string` be used instead of `char*`?

Reply:
Yes, but those are string literals, so why would we?